src/execution_plan/execution_plan_build/execution_plan_construct.c

```diff
@@ -255,6 +255,10 @@ static void _buildForeachOp
 	// connect foreach op to its child operation
 	ExecutionPlan_AddOp(foreach, embedded_plan->root);
 
+	// introduce the list-components variable (alias) to the record-map of the
+	// main plan
+	OpBase_Modifies(foreach, exp->resolved_name);
+
```

Comment:
the "outer" plan and the loop's "inner" plan should be kept separated, each having their own record mapping, do note the outer plan mapping is a subset of the inner plan, e.g.:
outer mapping: `{a:0, b:1}`
inner mapping: `{a:0, b:1, c:2}`

As the inner plan is exposed to all variable in the outer scope but not vice versa.